app/middleware/middleware_healthcheck.rb

```diff
@@ -0,0 +1,32 @@
+class MiddlewareHealthcheck
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if env['PATH_INFO'.freeze] == '/status'.freeze
+      response = [200, { 'Content-Type' => 'application/json' }]
+
+      # checks the database health
+      begin
+        db_healthy = ActiveRecord::Migrator.current_version != 0
+      rescue StandardError => e
+        Rails.logger.info "Database error: #{e}"
+        db_healthy = false
+      end
+
+      # checks the health of launchpad
+      cmr_client = Cmr::Client.client_for_environment(Rails.configuration.cmr_env, Rails.configuration.services)
+      launchpad_healthy = cmr_client.launchpad_healthcheck.body == 'OK'.freeze
+      response[2] = ["{\"database\": #{db_healthy}, \"launchpad\": #{launchpad_healthy}}"]
+
+      # If launchpad is disabled then we will not report a 500 error if launchpad still fails
+      unless (ENV['launchpad_login_required'] != 'true' || launchpad_healthy) && db_healthy
```

Comment:
I think you should change this to use `if`, it will be easier to be sure what we are checking for.
http://www.railstips.org/blog/archives/2008/12/01/unless-the-abused-ruby-conditional/

Reply:
Also, if we don't need to check on launchpad if launchpad login is not required, then we should completely skip the call.

Reply:
I talked with Catalino about this. He said that in this case he would still like to do the launchpad healthcheck but not return a 503 error if the check fails.